presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -561,13 +563,18 @@ private SliceReadFunction arrayAsJsonReadFunction(ConnectorSession session, Colu
             // read array into a block
             Block block = ((BlockReadFunction) readFunction).readBlock(resultSet, columnIndex);
 
-            // cast block to JSON slice
+            // convert block to JSON slice
+            Type blockType = ((ArrayType) type).getElementType();
+            List<Object> values = new ArrayList<>();
+            for (int position = 0; position < block.getPositionCount(); position++) {
```

Comment:
Maybe we can just:
```
type.getObjectValue(session, singleValueBlock(block), 0);
```

instead of doing this for each element separately?

Reply:
`ArrayType.getObjectValue()` ends up doing the same thing, but I suppose it's a bit cleaner to do it this way (although less efficient).

Reply:
I think in this case readability is more important, so whatever is terse and readable... up to you.